src/items/functions.md

```diff
@@ -148,30 +147,34 @@ extern "stdcall" fn new_i32_stdcall() -> i32 { 0 }
 ```
 
 Just as with [external block], when the `extern` keyword is used and the `"ABI` 
-is omitted, the ABI used defaults to `"C"`. That is, this
+is omitted, the ABI used defaults to `"C"`. That is, this:
 
 ```rust
 extern fn new_i32() -> i32 { 0 }
 let fptr: extern fn() -> i32 = new_i32;
 ```
 
-is identical to
+is equivalent to:
 
 ```rust
 extern "C" fn new_i32() -> i32 { 0 }
 let fptr: extern "C" fn() -> i32 = new_i32;
 ```
 
-Since functions with an ABI that differs from `"Rust"` do not support 
-unwinding in the exact same way that Rust does, unwinding past the end 
-of functions with such ABIs causes the process to abort. In LLVM, this is
-implemented by executing an illegal instruction.
+Functions with an ABI that differs from `"Rust"` do not support unwinding in the
+exact same way that Rust does. Therefore, unwinding past the end of functions
+with such ABIs causes the process to abort.
 
-Some ABIs that are identical to `"Rust"` are:
+**Non-normative note**: The LLVM backend of the current Rust implementation
+aborts the process by executing an illegal instruction.
 
-* `"rust-call"`
-* `"platform-intrinsic"`
-* `"rust-intrinsic"`
+**Non-normative note**: There are other ABIs available in unstable Rust that are
+equivalent to the `"Rust"` ABI, e.g.,
```

Comment:
Are they actually equivalent to `extern "Rust"`, though? I know for example that pointers to `extern "rust-intrinsic"` functions cannot be formed, since they have nowhere to point to.

Reply:
I see your point. The pointers can be formed (e.g. Rust programs that do this type check), but the compiler ICEs (this detects bugs in libcore, etc.). 

Maybe you are interested in [this comment](https://github.com/rust-lang-nursery/reference/pull/652#discussion_r314206468).

Reply:
I always assumed that because intrinsics are always inlined into the caller, they don't have their own distinct addresses and it's meaningless to speak of pointers to them. The compiler ICEs because it cannot generate code which forms such pointers, even when all types check out; in other words, it's not a deliberate feature (see also rust-lang/rust#15694). Am I wrong here?

Reply:
I'm not sure. I don't know what happens when you create a function pointer to, e.g., `llvm.assume` (does one get an LLVM error?), but for something like `core::intrinsic::sqrtf32` which emits `llvm.sqrt` creating a function pointer to it should work just fine. I'm not sure, but it might depend on the intrinsic.